### ISSUE TITLE
[CI] org.elasticsearch.indices.recovery.IndexRecoveryIT.testTransientErrorsDuringRecoveryAreRetried

### ISSUE THREAD

**Issue by original-brownbear** [elastic/elasticsearch#55863]
Failed for master here: https://gradle-enterprise.elastic.co/s/vajv4gmrdw5ba/console-log/raw

```
REPRODUCE WITH: ./gradlew ':server:integTest' --tests "org.elasticsearch.indices.recovery.IndexRecoveryIT.testTransientErrorsDuringRecoveryAreRetried" -Dtests.seed=2FB38A4D5D3AD00 -Dtests.security.manager=true -Dtests.locale=mk-MK -Dtests.timezone=America/Maceio -Dcompiler.java=14 -Druntime.java=11
  2> java.lang.AssertionError: timed out waiting for green state
        at __randomizedtesting.SeedInfo.seed([2FB38A4D5D3AD00:6B11D4FB13D54594]:0)
        at org.junit.Assert.fail(Assert.java:88)
        at org.elasticsearch.test.ESIntegTestCase.ensureColor(ESIntegTestCase.java:896)
        at org.elasticsearch.test.ESIntegTestCase.ensureGreen(ESIntegTestCase.java:849)
        at org.elasticsearch.test.ESIntegTestCase.ensureGreen(ESIntegTestCase.java:838)
        at org.elasticsearch.indices.recovery.IndexRecoveryIT.testTransientErrorsDuringRecoveryAreRetried(IndexRecoveryIT.java:819
```

**Comment by tbrooks8** [comment]
Does not reproduce for me. I ran it 1,000 times locally. I opened a PR to improve this test in various ways and add pertinent logging (https://github.com/elastic/elasticsearch/pull/55867).

**Comment by tbrooks8** [comment]
As and FYI the manner in which this test failure meant that multiple attempts to start the recovery happened BEFORE the disruption (no disruption logs occurred).

**Comment by tbrooks8** [comment]
Fixed by #55867.
